packages/react-native-bpk-component-calendar/src/BpkCalendar.ios.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  type StyleObj,
+  Button,
+  StyleSheet,
+  View,
+  ViewPropTypes,
+  requireNativeComponent,
+  NativeModules,
+} from 'react-native';
+
+const RCTCalendarView = requireNativeComponent('RCTCalendarView');
+
+const styles = StyleSheet.create({
+  base: {
+    width: '100%',
+    height: '100%',
+  },
+});
+
+export type Props = {
+  style: ?StyleObj,
+};
+
+const getDateInOneYear = () => {
+  const date = new Date();
+  const newMonth = date.getMonth() + 12;
+  date.setMonth(newMonth);
+  return date;
+};
+
+const BpkCalendar = props => {
+  const { onDateSelectionChanged, style: userStyle, ...rest } = props;
+
+  const style = [styles.base];
+  if (userStyle) {
+    style.push(userStyle);
+  }
+
+  return (
+    <View style={style} {...rest}>
+      <RCTCalendarView
+        locale="en-GB"
+        onDateSelection={event => {
```

Comment:
I _think_ it's more efficient to define this outside of the render block so that the function doesn't get recreated every time the component re-renders. @tiagoengel knows more about this than me though.

Reply:
The problem with this is that `onDateSelection` is always changing, so even if nothing really changed this will cause the react to interpret this as a new prop and re render the three. This is normally not a big unless it is part of a big three and it causes it to re-render. 

Not sure how much this affects things here as it is going directly to the 'bridged' component, plus the way things are now it's not possible to not do it. 

In order to avoid creating a new function every time we would have to change change `onDateSelection` accept `onDateSelectionChanged ` and return a "memoized" function that only receives the event. Something like this:

```
const onDateSelection = memoize(callback => {
    return (event) => {
       // do stuff
     }
}```

So for the same `onDateSelectionChanged` callback we would always have the same function.

But again this might not be required here so I wouldn't hold this PR because of this.